src/cc65/coptptrload.c

```diff
@@ -1457,3 +1457,219 @@ unsigned OptPtrLoad17 (CodeSeg* S)
+**      ldx     #0
+**      and     #mask          (any value < 0x80)
+**      jsr     aslax1/shlax1
+**      clc
+**      adc     #<(label+0)
+**      tay
+**      txa
+**      adc     #>(label+0)
+**      tax
+**      tya
+**      ldy     #$01
+**      jsr     ldaxidx
+**
+** and replace it by:
+**
+**      and     #mask          (remove if == 0x7F)
+**      asl
+**      tay
+**      lda     label,y
+**      ldx     label+1,y
+*/
+{
+    unsigned Changes = 0;
+    unsigned I;
+
+    /* Walk over the entries */
+    I = 0;
+    while (I < CS_GetEntryCount (S)) {
+
+        CodeEntry* L[12];
+
+        /* Get next entry */
+        L[0] = CS_GetEntry (S, I);
+
+        /* Check for the sequence */
+        if (L[0]->OPC == OP65_LDX                               &&
+            CE_IsKnownImm(L[0], 0)                              &&
+            CS_GetEntries (S, L+1, I+1, 11)                     &&
+            L[1]->OPC == OP65_AND                               &&
+            L[1]->AM == AM65_IMM                                &&
+            CE_HasNumArg (L[1]) && L[1]->Num <= 0x7F            &&
+            L[2]->OPC == OP65_JSR                               &&
+            L[3]->OPC == OP65_CLC                               &&
+            L[4]->OPC == OP65_ADC                               &&
+            L[5]->OPC == OP65_TAY                               &&
+            L[6]->OPC == OP65_TXA                               &&
+            L[7]->OPC == OP65_ADC                               &&
+            L[8]->OPC == OP65_TAX                               &&
+            L[9]->OPC == OP65_TYA                               &&
+            L[10]->OPC == OP65_LDY                              &&
+            CE_IsKnownImm(L[10], 1)                             &&
+            L[4]->Arg[0] == '<'                                 &&
+            L[7]->Arg[0] == '>'                                 &&
+            strlen(L[4]->Arg) > 3                               &&
+            strlen(L[7]->Arg) > 3                               &&
+            strcmp(L[4]->Arg+1, L[7]->Arg+1) == 0               &&
+            (strcmp (L[2]->Arg, "aslax1") == 0          ||
+             strcmp (L[2]->Arg, "shlax1") == 0)                 &&
+            CE_IsCallTo (L[11], "ldaxidx")                      &&
+            !CS_RangeHasLabel (S, I+1, 11)) {
+
+            CodeEntry* X;
+            char* Label;
+            int Len = strlen(L[4]->Arg);
+
+            /* Track the insertion point */
+            unsigned IP = I + 12;
+
+            /* asl a */
+            X = NewCodeEntry (OP65_ASL, AM65_ACC, "a", 0, L[2]->LI);
+            CS_InsertEntry (S, X, IP++);
+
+            /* tay */
+            X = NewCodeEntry (OP65_TAY, AM65_IMP, 0, 0, L[2]->LI);
+            CS_InsertEntry (S, X, IP++);
+
+            /* allocate Label memory */
+            Label = xmalloc (Len);
+            /* lda label,y */
+            Label = memcpy (Label, L[4]->Arg+2, Len-3);
```

Comment:
I like the style that you had here before.  `xmalloc()` was the first argument of `memcpy()`; and, `Label` was assigned from the return value of `memcpy()`.  It's compact.
[Also, it's the way that `OptPtrLoad3()` and `OptPtrLoad4()` use `xmalloc()`.]